massgov/openmass
# DP-22132: Create reusable architecture for tagging content with collections, topics and subtopics

**Description:**
Creating a reusable mechanism for a collections feature, starting with a new Collections vocabulary, EOTSS Service Catalog terms, and a multiple term reference field on the How-to content type. The Term Reference Tree module has also been installed as a field widget for selecting Collection terms.


**Jira:** (Skip unless you are MA staff)
https://jira.mass.gov/browse/DP-22132


**To Test:**
- Login as an Administrator or Content Administrator and go to /admin/content and filter by "How-to" content type.
- Click the Edit button for a result.
- Scroll to the bottom of the page and verify there is a new Collections field.
- [ ] Verify you can expand the term options and select terms at all level.
- [ ] Verify the field help text is... well, helpful.
- Go to /admin/structure/taxonomy/manage/collections/overview.
- Click the edit button for a term.
- [ ] Verify there is a Landing Page link field and it works as expected.
- Login as a user with a role that is not Administrator or Content Administrator. (eg. Author)
- Create or edit a How-to node.
- [ ] Verify you cannot see the Collections field.


**Screenshots/GIFs:**
![collections_field](https://user-images.githubusercontent.com/1434979/119906646-9aadc100-bf1c-11eb-95a4-77e4892e8b4f.png)

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)
